src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -222,5 +234,13 @@ private static JObject ReadJson(string packageSpecPath)
 
             return json;
         }
+
+        private static HashSet<RestoreOutputType> RestorableTypes = new HashSet<RestoreOutputType>()
```

Comment:
Can this be a project's public attribute `IsRestorable`? It'd make the statement at line 141 more readable.